ddl/db_test.go

```diff
@@ -991,7 +991,11 @@ func TestAddIndexFailOnCaseWhenCanExit(t *testing.T) {
 	tk.MustExec("drop table if exists t")
 	tk.MustExec("create table t(a int, b int)")
 	tk.MustExec("insert into t values(1, 1)")
-	tk.MustGetErrMsg("alter table t add index idx(b)", "[ddl:-1]DDL job rollback, error msg: job.ErrCount:1, mock unknown type: ast.whenClause.")
+	if variable.DDLEnableDistributeReorg.Load() {
+		tk.MustGetErrMsg("alter table t add index idx(b)", "[ddl:-1]job.ErrCount:0, mock unknown type: ast.whenClause.")
```

Comment:
Compared with disabling "DDLEnableDistributeReorg", why is it not rolling back?

Reply:
Dist-reorg is relatively complex and will be optimized later
I added todo https://github.com/pingcap/tidb/pull/40542/files#diff-59a1574ea9d9b6b78f0bd21d69e88f3861fae42835b177362638bea861a8397eR997